src/tqec/computation/block_graph.py

```diff
@@ -414,6 +414,57 @@ def from_dae_file(filename: str | pathlib.Path, graph_name: str = "") -> BlockGr
 
         return read_block_graph_from_dae_file(filename, graph_name)
 
+    def relabel_cubes(
+        block_graph: BlockGraph, identifier: Position3D | str, new_label: str
```

Comment:
According to the method name, I would rather see the following interface:

```suggestion
        block_graph: BlockGraph, label_mapping: Mapping[Position3D | str, str]
```

Reply:
> According to the method name, I would rather see the following interface:

Yep, sorry. I was thinking on relabel more than one cube depending on the label, but I didn't realize I should name it `relabel_cube`, so yes, your solution make sense :)

> A few comments, the main one being about the interface. I think it would be beneficial to be able to change multiple labels (or all of them) with one call to the method, hence the proposed interface. Let me know what you think about it.

The purpose was to allow to relabel more than one cube, especially in case of ZXCubes, because the ones that are not a node in the ZXGraph, I think normally will be labeled with a simple name. For cubes that doesn't have label, you can rename all of them at once `block_graph.relabel_cubes("", "Cubes")` but for Ports always has to be one by one because they are unique labels. Remember that I'm a really beginner at this, so anything is good for you, it's going to be good for me :)

Reply:
> block_graph: BlockGraph, label_mapping: Mapping[Position3D | str, str]

Now I see, sorry. Definitely yes, this allows all at once, perfect